app/src/main/java/ch/epfl/sdp/musiconnect/MapsActivity.java

```diff
@@ -236,22 +301,33 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         }
     }
 
+    //fetches users coordinates, updates them, and save them to cache
+    //(right now, only creates random nearby position and saves user to cache until database is implemented)
+    private void updateUsers(){
+        if(setLoc == null){             //Might be called before we get the first update to the location;
+            return;
+        }
 
-    private void updateProfileList() {
-        profiles.clear();
+        Random random = new Random();
 
-        Iterator<Marker> iter = markers.iterator();
-        while (iter.hasNext()) {
-            Marker marker = iter.next();
-            marker.remove();
-            iter.remove();
+        for(Musician m:allUsers){
+            double lat = setLoc.getLatitude() + (((double)random.nextInt(5)-2.5) /100);
+            double lng = setLoc.getLongitude() + (((double)random.nextInt(5)-2.5) /100);
+            m.setLocation(new MyLocation(lat,lng));
         }
 
+        saveUsersToCache();
+    }
 
-        if (setLoc == null) {
+    //From the users around the area, picks the ones that are within the threshold distance.
+    private void updateProfileList() {
+        if(setLoc == null){             //Might be called before we get the first update to the location;
             return;
+        } else{
+            delay = 20000;              //sets a 20 sec long delay on updates when everything is in place
```

Comment:
You have a lot of magic numbers in this file, try to extract them out into constants.